src/azul/project/hca/transformers.py

```diff
@@ -369,16 +419,30 @@ def transform(self,
                             manifest=manifest,
                             metadata_files=metadata_files)
         project = self._get_project(bundle)
-        for specimen in bundle.specimens:
+        samples: MutableMapping[str, Sample] = dict()
+        for si in bundle.sequencing_input:
+            self.get_ancestor_samples(samples, si)
+        for sample in samples.values():
             visitor = TransformerVisitor()
-            specimen.accept(visitor)
-            specimen.ancestors(visitor)
-            contents = dict(specimens=[_specimen_dict(specimen, visitor.visited_entities)],
+            sample.accept(visitor)
+            sample.ancestors(visitor)
+            contents = dict(samples=[{'sample_type': api.schema_names[type(sample)],
+                                      'document_id': str(sample.document_id),
```

Comment:
There seems to be some redundance here. The document and biomaterial IDs are tracked in the concrete inner entities as well. I would like to discuss this in person.